### ISSUE TITLE
Filename extension specification

### ISSUE THREAD

**Issue by ghost** [AppImage/AppImageSpec#12]
I noticed the draft says:

```
MUST NOT rely on any specific file name extension, although it is RECOMMENDED that the file name extension .AppImage is used whenever a file name extension is desired. Futher it is RECOMMENDED to follow the naming scheme ApplicationName-$VERSION-$ARCH.AppImage in cases in which it is desired to convey this information in the file name
```

To some like me the AppImage makes the filename longer than necessary and, subjectively, awkward to read. For example if I were to package Gedit, it would look like `Gedit.AppImage`, the extension being longer and more "noisy" than the application's name.

Is it possible to establish a shorter extension as an alternative? The conventional 3-4 character extension? Maybe `.app`?

**Comment by probonopd** [comment]
Users are free to move and rename as they whish, e.g., could even copy the AppImage to `/usr/local/bin/gedit` and have it on the `$PATH` this way. The filename extension is in part to make the format known.

**Comment by ghost** [comment]
I understand and I think what would you are proposing will generally work.

But with the absence of a conventional short extension, packages will either result to the full `AppImage` extension or invent their own short names (as you say). This can lead to difficulties with tools like `appimaged`; they will have to result in sophisticated approaches to selecting AppImage packages over generic executables and other files. Also think of typing an `mv *.Appimage` or `rm *.Appimage` command.

In short I think short extensions have merits and it's probably better to pre-empt separate attempts with a recommendation. (people can choose to ignore those)

**Comment by probonopd** [comment]
`appimaged` does not rely on AppImage filename extensions, nor should anything else. Software should check the magic numbers instead as per the AppImageSpec. So, the filename extension is purely a cosmetic thing and not really needed for functionality.

**Comment by pdfkungfoo** [comment]
> *"Software should check the magic numbers instead as per the AppImageSpec."*

BTW, do recent versions of the *`file`* utility already discover AppImage files? (My newest one, running on an openSUSE Tumbleweed is *`file-5.32`*, which does not identify AppImages other than as ELF binaries).

Has anybody already thought of submitting a patch to the *`file`* maintainers?
Has anybody created definitions for local *magic* files to patch with, so they can now identify AppImages?

**Comment by probonopd** [comment]
I think AppImage type-1 and type-2 file formats have been submitted to [shared-mime-info](https://freedesktop.org/wiki/Software/shared-mime-info/) but I don't think the `magic` command uses that. But I'm not sure!
